viadot/tasks/customer_gauge.py

```diff
@@ -131,23 +387,36 @@ def run(
             end_date=end_date,
         )
         cur = customer_gauge.get_cursor(json_data)
-        df = customer_gauge.to_df(json_data)
-        df_list.append(df)
+
+        jsn = self.get_data(json_data)
+        total_json += jsn
+
         if total_load == True:
             if cursor is None:
                 logger.info(
-                    f"Downloading all the data from the {self.endpoint or self.endpoint_url} endpoint. Process might take a few minutes..."
+                    f"Downloading all the data from the {self.endpoint or self.endpoint_url} endpoint." 
+                    "Process might take a few minutes..."
                 )
             else:
                 logger.info(
                     f"Downloading starting from the {cursor} cursor. Process might take a few minutes..."
                 )
-            while df.empty == False:
+            while jsn:
                 json_data = customer_gauge.get_json_response(cursor=cur)
                 cur = customer_gauge.get_cursor(json_data)
-                df = customer_gauge.to_df(json_data)
-                df_list.append(df)
+                jsn = self.get_data(json_data)
+                total_json += jsn
```

Comment:
Are you sure, you are not creating infinitive loop here? As variable jsn is assigning with every iteration same data from json_data varieble. Maybe try to use for loop or add some breakouts section inside while loop to avoid infinite iteration.

Reply:
last api call will return empty 'data', this is why `jsn = self.get_data(json_data)` will be empty and the loop will stop.